File: src/main/java/io/playground/orderservice/application/saga/port/client/PaymentClientPort.java
package io.playground.orderservice.application.saga.port.client;

import io.playground.orderservice.application.saga.dto.ClientDto;

public interface PaymentClientPort {
    void approvePayment(ClientDto.ApprovePaymentRequest request);

    void cancelPayment(ClientDto.CancelPaymentRequest request);

    void cancelPartially(ClientDto.CancelPartiallyRequest request);
}

File: src/main/kotlin/io/playground/orderservice/application/saga/port/client/PaymentClientPort.kt
package io.playground.orderservice.application.saga.port.client

import io.playground.orderservice.application.saga.dto.ClientDto

interface PaymentClientPort {
    fun approvePayment(request: ClientDto.ApprovePaymentRequest)

    fun cancelPayment(request: ClientDto.CancelPaymentRequest)

    fun cancelPartially(request: ClientDto.CancelPartiallyRequest)
}
